### PR TITLE
PrimeFactors: Improve triggering

### LINKED ISSUE #2799: PrimeFactors: Improve triggering
Should trigger on "is 83 prime"

---

IA Page:  http://duck.co/ia/view/prime_factors 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @austinheimark

### PR DESCRIPTION
**Pull request by darsnack**
## Description of changes
Previously, PrimeFactors could not detect a trigger of the form "is 83 prime". The following changes were made to fix this issue:
- Added check to regex guard if query matches form "is <number> prime"
- Strips all preceeding non-digit characters from remainder
- Includes test cases for new trigger

## Related Issues and Discussions
Fixes #2799

## People to notify
@moollaza @stevenmg @austinheimark 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [x] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [x] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/prime_factors

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/PrimeFactors.pm [hunk 1]
```diff
@@ -22,6 +22,7 @@ triggers startend => (
     'prime factorization of',
     'factorize',
     'prime factorize',
+    'prime'
```
Comment (moollaza) [inline lib/DDG/Goodie/PrimeFactors.pm]: let's set this to an `end` trigger:

```
triggers start => ( ... );
triggers end => 'prime', 'prime number';
```

File: lib/DDG/Goodie/PrimeFactors.pm [hunk 2]
```diff
@@ -88,9 +89,12 @@ sub format_answer {
 }
 
 handle remainder => sub {
-    # Exit if it's not a digit.
+    # Exit if it's not a digit or if it doesn't match the form "is <number> prime"
     # TODO: We should accept different number formats.
-    return unless /^\d+$/;
+    return unless (/^\d+$/ || ($req->query =~ /^is\s\d+\sprime$/i));
```
Comment (moollaza) [inline lib/DDG/Goodie/PrimeFactors.pm]: we should allow for queries like `is 4 prime`, `is 4 a prime`, `is 4 prime number`, and `is 4 a prime number`

**Comment by darsnack** [comment]
@moollaza I made the requested changes. It's been a couple of days, so I thought I'd ping you for a review.

I'm new to PRs on Github. Do I need to ping the necessary people so they know I have made a new commit that fixes issues raised?

File: lib/DDG/Goodie/PrimeFactors.pm [hunk 3]
```diff
@@ -91,7 +97,7 @@ sub format_answer {
 handle remainder => sub {
     # Exit if it's not a digit or if it doesn't match the form "is <number> prime"
     # TODO: We should accept different number formats.
-    return unless (/^\d+$/ || ($req->query =~ /^is\s\d+\sprime$/i));
+    return unless (/^\d+$/ || ($req->query =~ /^is\s\d+\s(a\s)?prime(\snumber)?$/i));
```
Comment (moollaza) [inline lib/DDG/Goodie/PrimeFactors.pm]: fwiw you don't have to be explicit about the `\s`: `/^is \d+ (a )?prime( number)?$/i`

Also if we `handle query_lc` we don't need to worry about using case insensitive regex 👍

**Comment by moollaza** [comment]
> Do I need to ping the necessary people so they know I have made a new commit that fixes issues raised?

Yes please. I no longer subscribe to commit notification because there was just too much flooding my inbox

**Comment by moollaza** [comment]
@darsnack Thanks! This LGTM :+1:

**Pull request merged**